lib/plugins/aws/package/compile/events/s3/configSchema.js

```diff
@@ -0,0 +1,813 @@
+    StorageClass: {
+      enum: [
+        'DEEP_ARCHIVE',
+        'GLACIER',
+        'INTELLIGENT_TIERING',
+        'ONEZONE_IA',
+        'OUTPOSTS',
+        'REDUCED_REDUNDANCY',
+        'STANDARD',
+        'STANDARD_IA',
+      ],
+    },
+  },
+  required: ['Bucket'],
+  additionalProperties: false,
+  dependencies: {
+    AccessControlTranslation: ['Account'],
+  },
+};
+
+const ReplicationRuleAndOperator = {
+  type: 'object',
+  properties: {
+    Prefix: {
+      type: 'string',
+    },
+    TagFilters: {
+      type: 'array',
+      items: TagFilter,
+    },
+  },
+  required: [],
+  additionalProperties: false,
+};
+
+const ReplicationRuleFilter = {
+  type: 'object',
+  properties: {
+    And: ReplicationRuleAndOperator,
+    Prefix: {
+      type: 'string',
+    },
+    TagFilter,
+  },
+  required: [],
+  additionalProperties: false,
+};
+
+const ReplicationRule = {
+  type: 'object',
+  properties: {
+    DeleteMarkerReplication,
+    Destination: ReplicationDestination,
+    Filter: ReplicationRuleFilter,
+    Id: {
+      type: 'string',
+      maxLength: 255,
+    },
+    Prefix: {
+      type: 'string',
+    },
+    Priority: {
+      type: 'integer',
+    },
+    SourceSelectionCriteria,
+    Status: {
+      enum: ['Disabled', 'Enabled'],
+    },
+  },
+  required: ['Destination', 'Status'],
+  additionalProperties: false,
+};
+
+const ReplicationConfiguration = {
+  type: 'object',
+  properties: {
+    Role: { $ref: '#/definitions/awsArn' },
+    Rules: {
+      type: 'array',
+      items: ReplicationRule,
+      minItems: 1,
+      maxItems: 1000,
+    },
+  },
+  required: ['Role', 'Rules'],
+  additionalProperties: false,
+};
+
+const RoutingRuleCondition = {
+  type: 'object',
+  properties: {
+    HttpErrorCodeReturnedEquals: {
+      type: 'string',
+    },
+    KeyPrefixEquals: {
+      type: 'string',
+    },
+  },
+  required: [],
+  additionalProperties: false,
+};
+
+const RoutingRule = {
+  type: 'object',
+  properties: {
+    RedirectRule,
+    RoutingRuleCondition,
+  },
+  required: ['RedirectRule'],
+  additionalProperties: false,
+};
+
+const Tag = {
+  type: 'object',
+  properties: {
+    Key: {
+      type: 'string',
+    },
+    Value: {
+      type: 'string',
+    },
+  },
+  required: ['Key', 'Value'],
+  additionalProperties: false,
+};
+
+const VersioningConfiguration = {
+  type: 'object',
+  properties: {
+    Status: {
+      enum: ['Enabled', 'Suspended'],
+    },
+  },
+  required: ['Status'],
+  additionalProperties: false,
+};
+
+const WebsiteConfiguration = {
+  type: 'object',
+  properties: {
+    ErrorDocument: {
+      type: 'string',
+    },
+    IndexDocument: {
+      type: 'string',
+    },
+    RedirectAllRequestsTo,
+    RoutingRules: {
+      type: 'array',
+      items: RoutingRule,
+    },
+  },
+  required: [],
+  additionalProperties: false,
+};
+
+module.exports = {
+  type: 'object',
+  properties: {
+    accelerateConfiguration: AccelerateConfiguration,
```

Comment:
Let's keep not reusable parts inline, e.g. in this case `AccelerateConfiguration` is defined only here.

It's easier for inspection to have it in-place, therefore I would seclude to local vars only if there's a reuse use case